File: web/domain/filing/models/deductions/loan/deduction-80eeb-model.ts
/**
 * Deduction 80EEB Interface - Electric Vehicle Loan Interest
 * Mirrors: section_80eeb in FilingModel
 * Extends Deduction80EModel with vehicle details
 */

import { Deduction80EModel } from '../../deduction-80e/models/deduction-80e-model'

export interface Deduction80EEBModel extends Deduction80EModel {
  vehicleMakeModel: string
  vehicleRegistrationNumber: string
}

File: api/filing/models/deduction_80eeb_model.py
from .deduction_80e_model import Deduction80EModel


class Deduction80EEBModel(Deduction80EModel):
    vehicle_make_model: str | None = None
    vehicle_registration_number: str | None = None

    @property
    def itr_vehicle_reg_no(self) -> str:
        return ((self.vehicle_registration_number or "").strip() or "NA")[:50]
